src/components/plots/GeneSearchBar.jsx

```diff
@@ -25,31 +26,56 @@ const GeneSearchBar = (props) => {
 
   const [options, setOptions] = useState([]);
 
-  // pass reactive component as value (search text) to allow auto clear on select
   const [value, setValue] = useState('');
 
   const onOptionSelect = (newGene) => {
-    const genes = [...config?.selectedGenes, newGene];
-
-    onSelect(genes);
-    setValue('');
+    const genes = value.split(/(?<!-)[,\s]+(?!-)/);
+    genes.splice(-1, 1, `${newGene}, `);
+    setValue(genes.join(', '));
+    setOptions([]);
   };
 
-  const onSearch = (searchText) => {
-    setValue(searchText);
+  const onSearch = (input) => {
+    setValue(input);
+
+    const genes = input.split(/(?<!-)[,\s]+(?!-)/);
+    const searchText = genes[genes.length - 1];
+
     setOptions(!searchText ? [] : filterGenes(searchText, geneList, config?.selectedGenes));
   };
 
+  const onClick = () => {
```

Comment:
Let's rename this to something else - maybe `addGene`?. The naming pattern `on...` is usually used for callbacks that are passed into the component.